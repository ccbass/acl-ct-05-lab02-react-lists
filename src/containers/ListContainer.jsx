import React, { useEffect, useState } from 'react';
import { fetchAllHAChars } from '../services/arnoldAPI';
import { ArnoldList } from "../components/arnold/ArnoldList.jsx";

export function ListContainer() {
    const [ chars, setChars ] = useState(null)

    useEffect(async () => {
        const charData = await fetchAllHAChars()
        setChars(charData)
    }, [])

    return (
        <div>
            {!chars && <h1>LOADING</h1>}
            {chars && 
                <div>
                    <h3>Hey Arnold Characters Page!</h3>
                    <ArnoldList characters={chars}/>
                </div>
            }
        </div>
    )
}

