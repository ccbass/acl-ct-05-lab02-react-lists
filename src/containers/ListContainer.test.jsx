import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import { ListContainer } from './ListContainer';
import {BrowserRouter} from 'react-router-dom'

describe('ListContainer', () => {
  it('renders a list of HA characters on the page', async () => {
    render(<BrowserRouter><ListContainer /></BrowserRouter>);

    screen.getByText('LOADING');

    const welcome = await screen.findByText('Hey Arnold Characters Page!');
    expect(welcome).toMatchSnapshot();
    expect(welcome).toBeInTheDocument();

    // const list = await waitFor( () => screen.getByText('Hey Arnold Characters Page!') )

    // expect(await screen.findByText('Hey Arnold Characters Page!')).toBeInTheDocument()
  });
});