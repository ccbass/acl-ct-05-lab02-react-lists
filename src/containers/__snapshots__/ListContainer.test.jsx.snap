// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListContainer renders a list of HA characters on the page 1`] = `
<h3>
  Hey Arnold Characters Page!
</h3>
`;
